CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration.uc

```diff
@@ -0,0 +1,134 @@
+//---------------------------------------------------------------------------------------
+//  AUTHOR:  NotSoLoneWolf
+//  PURPOSE: This class contains a variety of tools
+//           for this mod and others to interact with
+//           the infiltration template system.
+//---------------------------------------------------------------------------------------
+//  WOTCStrategyOverhaul Team
+//---------------------------------------------------------------------------------------
+
+class X2Helper_Infiltration extends Object config(Infiltration);
+
+var config int PERSONNEL_INFIL;
+var config int PERSONNEL_DETER;
+
+var config array<int> RANKS_DETER;
+
+function int GetSquadInfiltration(array<StateObjectReference> Soldiers)
+{
+	local StateObjectReference	UnitRef;
+	local int					TotalInfiltration;
+
+	TotalInfiltration = 0;
+	foreach Soldiers(UnitRef)
+	{
+		TotalInfiltration += GetSoldierInfiltration(Soldiers, UnitRef);
+	}
+
+	return TotalInfiltration;
+}
+
+function int GetSoldierInfiltration(array<StateObjectReference> Soldiers, StateObjectReference UnitRef)
+{
+	local XComGameStateHistory		History;
+	local XComGameState_Unit		UnitState;
+	local array<XComGameState_Item>	CurrentInventory;
+	local XComGameState_Item		InventoryItem, MultiplierItem;
+	local int						UnitInfiltration;
+	local int						TotalInfil;
+	local name						ItemCategory, MultCategory;
+	local float						Multiplier;
+
+	local X2InfiltrationModTemplateManager InfilMgr;
+	InfilMgr = class'X2InfiltrationModTemplateManager'.static.GetInfilTemplateManager();
+
+	History = `XCOMHISTORY;
+
+	if(UnitRef.ObjectID <= 0)
+		return 0;
+
+	UnitState = XComGameState_Unit(History.GetGameStateForObjectID(UnitRef.ObjectID));
+
+	if(!UnitState.IsSoldier())
+		return default.PERSONNEL_INFIL;
+
+	UnitInfiltration = 0;
+
+	CurrentInventory = UnitState.GetAllInventoryItems();
+	foreach CurrentInventory(InventoryItem)
+	{
+		if(InventoryItem != none)
+		{
+			TotalInfil = InfilMgr.GetInfilTemplateFromItem(InventoryItem.GetMyTemplateName()).InfilModifier;
+			ItemCategory = InventoryItem.GetMyTemplate().ItemCat;
+
+			foreach CurrentInventory(MultiplierItem)
+			{
+				if(MultiplierItem != none)
+				{
+					MultCategory = InfilMgr.GetInfilTemplateFromItem(MultiplierItem.GetMyTemplateName()).MultCategory;
+					Multiplier = InfilMgr.GetInfilTemplateFromItem(MultiplierItem.GetMyTemplateName()).InfilMultiplier;
```

Comment:
I'm not seeing any `none` checks though...

Reply:
It literally says it in the code block you highlighted though?

Reply:
`InfilMgr.GetInfilTemplateFromItem(MultiplierItem.GetMyTemplateName())` can be none, then `….MultCategory` throws an accessed none warning. Same for the other instances of `GetInfilTemplateFromItem`.

Reply:
Alright, I've changed the none checks to look like this:
`(InfilMgr.GetInfilTemplateFromItem(InventoryItem.GetMyTemplateName()) != none)`
For both InventoryItem and MultiplierItem.